packages/jsii-pacmak/lib/targets/golang/documentation.ts

```diff
@@ -1,15 +1,73 @@
 import { Docs } from 'jsii-reflect';
-import { EmitContext } from './emit-context';
+import { Rosetta } from 'jsii-rosetta';
+import { CodeMaker } from 'codemaker';
 
 export class Documentation {
-  public constructor(private readonly docs: Docs) {}
+  public constructor(
```

Comment:
Changing this to accept Rosetta and CodeMaker in the constructor may limit the ability to change docs emit content based on context.

If we need to resolve type references to GoTypes in docs we may have to change this back in the future and make an instance of Documentation for each node in the AST that has doc strings.

I don't see type references within the jsii-reflect.Docs type that would warrant something like that but it's worth noting. Specifically for Rosetta we may need this functionality. I say we move forward with the current global Documentation instance until we have reason to reorganize, if we ever do.

Reply:
I was following the example in the [DotNetDocGenerator](https://github.com/aws/jsii/blob/master/packages/jsii-pacmak/lib/targets/dotnet/dotnetdocgenerator.ts#L26-L28). I was operating with the understanding that the Doc object was per jsii type -- we do currently have a situation in which a single jsii type translates into more than one GoType (e.g. classes -> struct + interface), which I chose to handle by having the interface have a custom docstring (See: https://github.com/aws/jsii/pull/1976/files#diff-3670bd67666fe33ab8b587ba294204abR147-R149) and then the original docs on the [struct](https://github.com/aws/jsii/pull/1976/files#diff-3670bd67666fe33ab8b587ba294204abR162). 

I'm not sure what other situations would require more context that would be stored in the EmitContext object -- is there a similar situation in other targets?

Reply:
Nope, can't find one after digging around. So this may be an unwarranted fear. Regardless with the emit logic encapsulated it wouldn't be a big change if for some reason we did need contextual doc emits.

Reply:
Yeah, I mean the `EmitContext` object was just holding the global codemaker and rosetta objects, which are both being used by the documenter -- didn't seem like rosetta would be used outside of doc generation, so I think it'll be okay.